logger/logger.go

```diff
@@ -9,7 +9,15 @@ import (
 )
 
 type TfLogger struct {
-	Name string
+	ctx context.Context
+}
+
+func NewTfLogger(ctx context.Context) *TfLogger {
+	return &TfLogger{ctx: ctx}
+}
+
+func SetTfLogger(tfLogger *TfLogger) {
```

Comment:
We could have another method such as SetTfLoggerFromContext which will get a new logger and set it but I think it will lead to a loss of readability of code.

SetTfLogger(NewTfLogger(ctx)) clearly say we get a new logger first and set it.